src/main/java/com/google/devtools/build/lib/bazel/repository/ProxyHelper.java

```diff
@@ -0,0 +1,99 @@
+package com.google.devtools.build.lib.bazel.repository;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import java.io.IOException;
+import java.net.Authenticator;
+import java.net.InetSocketAddress;
+import java.net.PasswordAuthentication;
+import java.net.Proxy;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * Helper class for setting up a proxy server for network communication
+ *
+ */
+public class ProxyHelper {
+
+  /**
+   * This method takes a proxyAddress as a String (ex. http://userId:password@proxyhost.domain.com:8000)
+   * and sets JVM arguments for http and https proxy as well as returns a java.net.Proxy object for optional use.
+   * 
+   * @param proxyAddress
+   * @return Proxy
+   * @throws IOException
+   */
+  @VisibleForTesting
```

Comment:
I wasn't sure what this annotation was for so I left it from when this method was in HttpDownloader.  Will remove.